pkg/server/field_mask_test.go

```diff
@@ -0,0 +1,280 @@
+		return updateFromMask(&src, &dst, m)
+	}
+
+	tests := []struct {
+		assertFunc func(t *testing.T, src, dst, result nestedStruct)
+		name       string
+		mask       model.Mask
+	}{
+		{
+			// Update one top-level field
+			name: "Update A int",
+			mask: []string{"A"},
+			assertFunc: func(t *testing.T, src, dst, result nestedStruct) {
+				// Check if field is updated
+				require.Equal(t, src.A, result.A)
+
+				// Reset field to dst value, and perform full check, to ensure that nothing else was changed
+				result.A = dst.A
+				assert.Equal(t, dst, result)
+			},
+		},
+		{
+			// Update two top-level fields
+			name: "Update A and B int and string",
+			mask: []string{"A", "B"},
+			assertFunc: func(t *testing.T, src, dst, result nestedStruct) {
+				require.Equal(t, src.A, result.A)
+
+				result.A, result.B = dst.A, dst.B
+				assert.Equal(t, dst, result)
+			},
+		},
+		{
+			// Update whole top-level struct
+			name: "Update C struct",
+			mask: []string{"C"},
+			assertFunc: func(t *testing.T, src, dst, result nestedStruct) {
+				require.Equal(t, src.C, result.C)
+
+				result.C = dst.C
+				assert.Equal(t, dst, result)
+			},
+		},
+		{
+			// Update top-level field of a nested struct
+			name: "Update C.D struct.float",
+			mask: []string{"C.D"},
+			assertFunc: func(t *testing.T, src, dst, result nestedStruct) {
+				require.Equal(t, src.C.D, result.C.D)
+
+				result.C.D = dst.C.D
+				assert.Equal(t, dst, result)
+			},
+		},
+		{
+			// Update a whole nested struct
+			name: "Update C.F struct.struct",
+			mask: []string{"C.F"},
+			assertFunc: func(t *testing.T, src, dst, result nestedStruct) {
+				require.Equal(t, src.C.F, result.C.F)
+
+				result.C.F = dst.C.F
+				assert.Equal(t, dst, result)
+			},
+		},
+		{
+			// Update slice of strings in a double nested struct (merge two slices)
+			name: "Update C.F.G struct.struct.[]string",
+			mask: []string{"C.F.G"},
+			assertFunc: func(t *testing.T, src, dst, result nestedStruct) {
+				// Check if all elements from src and dst are in result
+				for _, srcElem := range src.C.F.G {
+					require.Contains(t, result.C.F.G, srcElem)
+				}
+
+				for _, dstElem := range dst.C.F.G {
+					require.Contains(t, result.C.F.G, dstElem)
+				}
+
+				result.C.F.G = dst.C.F.G
+				assert.Equal(t, dst, result)
+			},
+		},
+		{
+			// Update slice of custom structs in a nested struct. (Merge two slices)
+			name: "Update C.H struct.struct.[]struct",
+			mask: []string{"C.H"},
+			assertFunc: func(t *testing.T, src, dst, result nestedStruct) {
+				// Check if all elements from src and dst are in result
+				for _, srcElem := range src.C.H {
+					require.Contains(t, result.C.H, srcElem)
+				}
+
+				for _, dstElem := range dst.C.H {
+					require.Contains(t, result.C.H, dstElem)
+				}
+
+				result.C.H = dst.C.H
+				assert.Equal(t, dst, result)
+			},
+		},
+		{
+			// Update map of strings in a nested field of struct. (Merge two maps)
```

Comment:
You have 'merge two maps' test case. 
Could you add a test case where a value for a specific key in the map is updated?

Reply:
Need implementation beforehand. For now can be ignored, as we don't have structs with maps

Reply:
hmm, we could return codes.Unimplemented.

Reply:
or we could ignore as we don't have maps in our updatable structs :)

Reply:
Well you already added case reflect.Map :)